src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -2349,28 +2344,21 @@ private int ScanText(out int start)
         private string GetString(int pos, int cch)
         {
             Debug.Assert(pos >= 0 && cch >= 0);
-            if (checked(pos + (cch * 2)) > _end)
+            if (checked(pos + (cch * sizeof(char))) > _end)
                 throw new XmlException(SR.Xml_UnexpectedEOF1, (string[]?)null);
             if (cch == 0)
                 return string.Empty;
-            // GetStringUnaligned is _significantly_ faster than unicode.GetString()
-            // but since IA64 doesn't support unaligned reads, we can't do it if
-            // the address is not aligned properly.  Since the byte[] will be aligned,
-            // we can detect address alignment my just looking at the offset
-            if ((pos & 1) == 0)
-                return GetStringAligned(_data, pos, cch);
-            else
-                return _unicode.GetString(_data, pos, checked(cch * 2));
-        }
 
-        private unsafe string GetStringAligned(byte[] data, int offset, int cch)
-        {
-            Debug.Assert((offset & 1) == 0);
-            fixed (byte* pb = data)
+            return string.Create(cch, (_data, pos), static (dstChars, state) =>
             {
-                char* p = (char*)(pb + offset);
-                return new string(p, 0, cch);
-            }
+                // bitblt source bytes directly into the destination char span
+                // n.b. source buffer assumed to be well-formed UTF-16 machine endian
```

Comment:
A bit of an expansion on this comment: the original code had two different behaviors based on whether the underlying data was aligned or unaligned. If the underlying data was aligned, the original code would basically memmove the contents of the old buffer into the new string, not performing any UTF-16 validation. If the underlying data was unaligned, the original code would go through `Encoding.Unicode.GetString`, which _would_ perform UTF-16 validation. This really only affects whether unpaired surrogate characters get replaced with `'\uFFFD'` in the destination string.

Since the updated code is just a simple memmove, I opted for the "don't perform any validation" behavior.